st2auth_ldap/ldap_backend.py

```diff
@@ -155,6 +155,12 @@ def authenticate(self, username, password):
             # Instantiate connection object and bind with service account.
             try:
                 connection = self._init_connection()
+
+                if self._bind_dn.find('{username}') != -1:
+                    self._bind_dn = self._bind_dn.format(username=username)
+                if self._bind_password.find('{password}') != -1:
```

Comment:
According to the `str.find()` documentation, you should not use it to check for the existence of a substring. See the note in the documentation for what you're supposed to use:
https://docs.python.org/3/library/stdtypes.html#str.find

We also don't even need to check for this. The `str.format` call will not error out if it doesn't find a format string, and return the string unchanged, so we don't need any sort of check at all.